### PR TITLE
Add support for CacheControl header

### LINKED ISSUE #18: Metadata: x-amz-meta-cache-control is set instead Cache-Control
Looks like a bug. The `x-amz-meta-cache-control` metadata key is set instead of `Cache-Control`. 

`.drone.yml`:

```
...
  s3_sync:
    acl: public-read
    region: $$AWS_REGION
    bucket: $$AWS_BUCKET
    access_key: $$AWS_ACCESS_KEY_ID
    secret_key: $$AWS_SECRET_ACCESS_KEY
    source: build/
    target: /
    delete: true
    metadata:
      "*.js":
        Cache-Control: "public, max-age: 10000000"
      "*.css":
        Cache-Control: "public, max-age: 10000000"
      "*.jpg":
        Cache-Control: "public, max-age: 10000000"
      "*.gif":
        Cache-Control: "public, max-age: 10000000"
      "*.png":
        Cache-Control: "public, max-age: 10000000"
      "*.html":
        Cache-Control: "public, max-age: 600"
```

### PR DESCRIPTION
**Pull request by brianfoshee**
This PR adds support for a `cache_control` key. 

Fixes #18

### CODE REVIEW

File: DOCS.md [hunk 1]
```diff
@@ -61,6 +63,9 @@ The `content_type` field the key is an extension including the leading dot `.`.
 In the  `content_encoding` field the key is an extension including the leading dot `.`. If you want to set a encoding type for files with no extension, set the key
 to th empty string `""`. If there are no matches for the `content_encoding` of a file, no content-encoding header will be added.
 
+In the  `cache_control` field the key is an extension including the leading dot `.`. If you want to set cahce control for files with no extension, set the key
```
Comment (nlf) [inline DOCS.md]: nit, `cahce` -> `cache`

File: DOCS.md [hunk 2]
```diff
@@ -61,6 +63,9 @@ The `content_type` field the key is an extension including the leading dot `.`.
...
+to th empty string `""`. If there are no matches for the `cache_control` of a file, no cache-control header will be added.
```
Comment (nlf) [inline DOCS.md]: nit, `to th empty` -> `to the empty`

**Review by nlf — CHANGES_REQUESTED** [review]

File: DOCS.md (hunk 2 above)
Comment (brianfoshee) [reply DOCS.md]: Fixed

File: DOCS.md (hunk 1 above)
Comment (brianfoshee) [reply DOCS.md]: fixed

**Comment by nlf** [comment]
LGTM

File: DOCS.md [hunk 3]
```diff
@@ -93,9 +98,7 @@ publish:
     delete: true
     metadata:
       "*.png":
-        Cache-Control: "max-age: 10000000"
```
Comment (tboerger) [inline DOCS.md:96]: No further cache control example anymore?

Comment (brianfoshee) [reply DOCS.md:96]: These didn't properly set the `Cache-Control` header - just `x-amz-meta-cache-control`. See line 49 for the new example.

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

**Comment by brianfoshee** [comment]
@nlf @tboerger is this good to be merged in?

**Comment by tboerger** [comment]
I have already approved, just waiting for a comment of @nlf

**Comment by nlf** [comment]
I commented before you :) I'm fine with it. On mobile right now, but someone else is free to merge

**Comment by brianfoshee** [comment]
@nlf @tboerger is there any way to get this rebuilt for docker hub? It looks like this image is old https://hub.docker.com/r/plugins/drone-s3-sync/

**Comment by nlf** [comment]
tried the one at https://hub.docker.com/r/plugins/s3-sync ?

**Comment by brianfoshee** [comment]
I have not - is that the same plugin, just a different image?

**Comment by nlf** [comment]
yup, some months ago plugins started getting pushed to different names

**Comment by brianfoshee** [comment]
great, thanks!

**Comment by brianfoshee** [comment]
@nlf I'm getting this error message using the new image: 

```
[info] Pulling image plugins/s3-sync:1.1
Must set access_key, secret_key, and bucket
```

The only change I made was adding `image: plugins/s3-sync:1.1` into the existing `s3_sync` section, which has those three keys set already.

Did something change regarding them?

Edit: are these 0.5-only now, or will they still work on 0.4?

**Comment by tboerger** [comment]
The master branch is for >= 0.5 only.

**Pull request merged**